docs/notebooks/analysis/nullpoint.ipynb

```diff
@@ -0,0 +1,235 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "9b3731e3-1936-4b26-8a05-2130669fd074",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "from pathlib import Path\n",
+    "\n",
+    "from plasmapy.analysis import nullpoint\n",
+    "\n",
+    "plt.rcParams[\"figure.figsize\"] = [10.5, 0.56 * 10.5]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eec0e81c-bbf8-4e01-b121-114401996f7c",
+   "metadata": {},
+   "source": [
+    "## Contents:\n",
+    "\n",
+    "1. [How Null Point Finder works](#How-Null-Point-Finder-works)\n",
+    "    1. [Locating a null point](#Locating-a-null-point)\n",
+    "    1. [Classifying a null point](#Classifying-a-null-point)\n",
+    "1. [Running through examples](#Running_through-examples)\n",
+    "    1. [Uniform regular grid with a model function](#Uniform-regular-grid-with-a-model-function)\n",
+    "    1. [Arbitary Regular Grid](#Arbitary-regular-grid)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "60702e11-905b-43b3-85ab-dc3f85bc9bab",
+   "metadata": {},
+   "source": [
+    "## How Null Point Finder works\n",
+    "Null point finder procides two functions that can locate and classify null point of a magnetic field. The first function is [**uniform_null_point_find()**](../../api/plasmapy.analysis.nullpoint.uniform_null_point_find.rst#uniform_null_point_find)  and the second one is [**null_point_find()**](../../api/plasmapy.analysis.nullpoint.null_point_find.rst#uniform_null_point_find). As the names suggest, `uniform_null_point_find()` is used to locate and classify the magnetic null points of magnetic field located within a regular grid with uniform spacing in each diemension. It requires the user to provide the spacing between grid points in each of the three dimensions in addition to the minimum and maximum coordinate in each dimentsion. Moreover, it requires the user to provide a function which generates the vector values at a given point. `uniform_null_point_find()` is useful for when the user knows of such a function that accurately models the strengths of the magentic field at each point in space. \n",
+    "On the other hand, `null_point_find()` is used when the user does not have an adequate modeling function. It also does not require the grid to have uniform spacing in each dimension. Instead, it will ask the user for three arrays (one for each dimension) of coordinates that determines the desired custom spacing in each of the three dimensions, and then constructs the resulting grid on its owns. Furthumore, it requires the user to input all of the three components of magnetic field values, each as a 3D array with the same size as the grid, with each entry representing the strength of that component of the magnetic field for the corresponding grid coordinate. \n",
```

Comment:
```suggestion
    "On the other hand, `null_point_find()` is used when the user does not have an adequate modeling function. It also does not require the grid to have uniform spacing in each dimension. Instead, it will ask the user for three arrays (one for each dimension) of coordinates that determines the desired custom spacing in each of the three dimensions, and then constructs the resulting grid on its own. Furthermore, it requires the user to input all of the three components of magnetic field values, each as a 3D array with the same size as the grid, with each entry representing the strength of that component of the magnetic field for the corresponding grid coordinate. \n",
```